src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -77,20 +85,100 @@ export const survivalClinicalDataNullValueSet = new Set([
+            } else {
+                dict[patientSurvival.months] = 1;
+            }
+            return dict;
+        },
+        {} as Dictionary<number>
+    );
+
     patientSurvivals.forEach((patientSurvival, index) => {
+        // we may see multiple data points at the same time
+        // only calculate the error for the last data point of the same months
+        if (
+            previousMonths === patientSurvival.months &&
+            patientSurvival.status
+        ) {
+            numberOfEvenrts++;
+        } else {
+            numberOfEvenrts = 1;
+        }
+        previousMonths = patientSurvival.months;
+        if (nEventDict[patientSurvival.months] === numberOfEvenrts) {
+            isErrorCalculatable = true;
+        } else {
+            isErrorCalculatable = false;
+        }
+        // only calculate standard error and confidence intervals for event data
         if (patientSurvival.status) {
             const atRisk = patientSurvivals.length - index;
             const estimate = previousEstimate * ((atRisk - 1) / atRisk);
             previousEstimate = estimate;
-            estimates.push(estimate);
+            // define calculate standard error and confidence intervals
+            // undefined is default value, means it's not found
+            let standardError: number | undefined = undefined;
+            let low95ConfidenceInterval: number | undefined = undefined;
+            let high95ConfidenceInterval: number | undefined = undefined;
+
+            if (isErrorCalculatable) {
+                // calculate standard error
+                // adjustedAtRisk need to add number of events back since we may have multiple events at the same time
+                const adjustedAtRisk =
+                    atRisk + nEventDict[patientSurvival.months] - 1;
+                const v1 =
+                    nEventDict[patientSurvival.months] /
+                    (adjustedAtRisk *
+                        (adjustedAtRisk - nEventDict[patientSurvival.months]));
+                const error = estimate * estimate * (previousError + v1);
+                previousError = previousError + v1;
+                standardError = Math.sqrt(error);
+
+                // calculate confidence intervals
+                const conficenceInterval = 0.95;
+                const zValue = jStat.normal.inv(
+                    1 - (1 - conficenceInterval) / 2,
+                    0,
+                    1
+                );
+                // Reference: https://github.com/cran/survival/blob/master/R/survfit.R#L308-L316
+                const xx = estimate === 0 ? NaN : estimate;
+                const se2 = (standardError / estimate) * zValue;
+                const temp1 = Math.exp(Math.log(xx) - se2);
+                const temp2 = Math.exp(Math.log(xx) + se2);
+                low95ConfidenceInterval = temp1;
+                high95ConfidenceInterval = Math.min(temp2, 1);
```

Comment:
based on log function used in survival package: https://github.com/cran/survival/blob/master/R/survfit.R#L308-L316

Reply:
if this stretch of code is logic that we've taken from somewhere else, maybe put in a helper funcion

Reply:
created a helper function